pkg/action/upgrade.go

```diff
@@ -70,6 +70,10 @@ func NewUpgrade(cfg *Configuration) *Upgrade {
 
 // Run executes the upgrade on the given release.
 func (u *Upgrade) Run(name string, chart *chart.Chart, vals map[string]interface{}) (*release.Release, error) {
+	if err := u.cfg.KubeClient.IsReachable(); err != nil {
```

Comment:
we should check if `u.DryRun` is. `helm upgrade --dry-run` does not rely on a cluster being connected IIRC

Reply:
https://github.com/helm/helm/blob/0afce196fda3bad3e7c30c4da53d8b18335524da/pkg/action/install.go#L157-L162

Reply:
scratch that... `--dry-run` != `--client-only`. It does rely on the cluster being reachable. Disregard